### PR TITLE
Round maxRequiredSize to match romSize calculation

### PR DESCRIPTION
**Pull request by theresa-m**
Fixes: https://github.com/eclipse/openj9/issues/5988
Fixes: https://github.com/eclipse/openj9/issues/7056
Fixes: https://github.com/eclipse/openj9/issues/5399

romSize was rounded here: https://github.com/eclipse/openj9/pull/5301

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins test sanity,extended osx jdk11

**Comment by pshipton** [comment]
Does this resolve #7056 ?

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/view/all/job/PullRequest-OpenJ9/1234/

**Comment by pshipton** [comment]
@theresa-m can you please look at the test failure, this isn't a failure I've seen in the nightly builds.
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_extended.functional_x86-64_mac_Personal/30/

**Comment by theresa-m** [comment]
I have verified locally that this fixes https://github.com/eclipse/openj9/issues/7056

Looking at the functional test failure..

**Comment by pshipton** [comment]
This should also fix #5399 so I've added it to the list in the description.

File: runtime/bcutil/ROMClassBuilder.cpp [hunk 1]
```diff
@@ -701,31 +701,15 @@ ROMClassBuilder::prepareAndLaydown( BufferManager *bufferManager, ClassFileParse
...
-				maxRequiredSize += bytesToPad;
-			}
-		}
+		/* Round up to sizeof(U_64).
+		*  
+		* Pad maxRequiredSize to the size to sizeof(U_64) in order to prevent memory corruption. 
+		* This mirrors ROM class padding in finishPrepareAndLaydown when the final ROM class size 
+		* is calculated.
+		*/
+		maxRequiredSize += (sizeof(U_64) - 1);
```
Comment (pshipton) [inline runtime/bcutil/ROMClassBuilder.cpp]: Could use the ROUND_UP_TO_POWEROF2 macro here.

**Comment by DanHeidinga** [comment]
There's an existing assert for this: https://github.com/eclipse/openj9/blob/9d35d06dcbf62bb83d05b93dc980e201f6aeae9e/runtime/bcutil/ROMClassBuilder.cpp#L761  Why didn't it catch this issue?

**Comment by pshipton** [comment]
Given the size of the file, there is likely a core file archived for the failures, like https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Test/Test_openjdk11_j9_extended.functional_x86-64_mac_Personal/30/functional_test_output.tar.gz

Probably doesn't matter, but the PR test failure occurred on osx1014-x86-1.
Running grinders I got some reproductions
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/501 - osx1013-x86-4
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/502 - osx1014-x86-2
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/503 - osx1014-x86-3
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/504 - osx1011-x86-2

Running grinders on last night's build without this change as well.

**Comment by pshipton** [comment]
Also failing with last night's build, so not introduced by this PR.
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/506
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/508
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/510

**Comment by pshipton** [comment]
Trc_BCU_Assert_True is level 3 and not enabled by default.
`TraceAssert=Trc_BCU_Assert_True NoEnv Overhead=1 Level=3 Assert="(P1)"`

**Comment by pshipton** [comment]
> How can we prevent this from occurring again?

We can add a test that enables the BCU asserts.

**Comment by DanHeidinga** [comment]
> We can add a test that enables the BCU asserts.

I'd rather enable the BCU asserts by default and measure the performance hit.  For reasons I don't yet understand, we're only seeing this issue with a particular class on OS X which a simple test wouldn't catch.  It needs to be the default to catch this in the general case.

**Comment by pshipton** [comment]
Created https://github.com/eclipse/openj9/issues/7140 for the extended.functional failure, which is now also seen in nightly builds.

**Comment by DanHeidinga** [comment]
> I'd rather enable the BCU asserts by default and measure the performance hit. For reasons I don't yet understand, we're only seeing this issue with a particular class on OS X which a simple test wouldn't catch. It needs to be the default to catch this in the general case.

I don't want this PR which fixes memory corruption to be held up by this but it is something we should address.  @theresa-m Can you open an issue for tracking changing the trace asserts to be level 1?  That change will need some startup performance testing to ensure it doesn't regress startup.

**Comment by theresa-m** [comment]
I've opened an issue https://github.com/eclipse/openj9/issues/7143 for enabling TraceAsserts by default. I've also updated the commit to use the ROUND_UP_TO_POWEROF2 macro.

File: runtime/bcutil/ROMClassBuilder.cpp [hunk 2]
```diff
@@ -701,31 +701,11 @@ ROMClassBuilder::prepareAndLaydown( BufferManager *bufferManager, ClassFileParse
...
-			UDATA bytesToPad = adjustedCount - maxRequiredSize;
-			if (0 != bytesToPad) {
-				maxRequiredSize += bytesToPad;
-			}
-		}
+		/* Pad maxRequiredSize to the size to sizeof(U_64) in order to prevent memory corruption. 
+		* This mirrors ROM class padding in finishPrepareAndLaydown when the final ROM class size 
+		* is calculated.
+		*/
+		ROUND_UP_TO_POWEROF2(maxRequiredSize, sizeof(U_64));
```
Comment (pshipton) [inline runtime/bcutil/ROMClassBuilder.cpp]: I think this needs to be `maxRequiredSize = ROUND_UP_TO_POWEROF2(maxRequiredSize, sizeof(U_64));`

**Comment by pshipton** [comment]
jenkins test sanity osx jdk11

**Pull request merged**